spyder/plugins/console/plugin.py

```diff
@@ -4,376 +4,129 @@
-        # Connecting the following signal once the dockwidget has been created:
-        self.shell.exception_occurred.connect(self.exception_occurred)
-        previous_crash = CONF.get('main', 'previous_crash', '')
-        if previous_crash:
-            self.exception_occurred(previous_crash, True,
-                                    is_faulthandler_report=True)
+    def exit_interpreter(self):
+        self.get_widget().exit_interpreter()
 
-    def exception_occurred(self, text, is_traceback, is_pyls_error=False,
-                           is_faulthandler_report=False):
-        """
-        Exception ocurred in the internal console.
-
-        Show a QDialog or the internal console to warn the user.
-        """
-        # Skip errors without traceback or dismiss
-        if (not is_traceback and self.error_dlg is None) or self.dismiss_error:
-            return
-
-        if CONF.get('main', 'show_internal_errors'):
-            if self.error_dlg is None:
-                self.error_dlg = SpyderErrorDialog(self)
-                self.error_dlg.set_color_scheme(CONF.get('appearance',
-                                                         'selected'))
-                self.error_dlg.close_btn.clicked.connect(self.close_error_dlg)
-                self.error_dlg.rejected.connect(self.remove_error_dlg)
-                self.error_dlg.details.go_to_error.connect(self.go_to_error)
-
-            if is_pyls_error:
-                title = "Internal Python Language Server error"
-                self.error_dlg.set_title(title)
-                self.error_dlg.title.setEnabled(False)
-            if is_faulthandler_report:
-                title = "Segmentation fault crash"
-                self.error_dlg.set_title(title)
-                self.error_dlg.title.setEnabled(False)
-                self.error_dlg.main_label.setText(
-                    _("<h3>Spyder crashed during last session</h3>"))
-                self.error_dlg.submit_btn.setEnabled(True)
-                self.error_dlg.steps_text.setText(
-                    _("Please provide any additional information you "
-                      "might have about the crash."))
-                self.error_dlg.set_require_minimum_length(False)
-            self.error_dlg.append_traceback(text)
-            self.error_dlg.show()
-        elif DEV or get_debug_level():
-            self.switch_to_plugin()
+    def execute_lines(self, lines):
+        self.get_widget().execute_lines(lines)
 
-    def close_error_dlg(self):
-        """Close error dialog."""
-        if self.error_dlg.dismiss_box.isChecked():
-            self.dismiss_error = True
-        self.error_dlg.reject()
+    def get_sys_path(self):
+        return self.get_widget().get_sys_path()
 
-    def remove_error_dlg(self):
-        """Remove error dialog."""
-        self.error_dlg = None
+    def handle_exception(self, text, is_traceback, is_pyls_error=False,
+                         is_faulthandler_report=False):

```

Comment:
Missing docstring.

Reply:
Fixed

Reply:
I don't see a docstring for this one yet. Please fix that.